File: clone-sprint2-3/fastfood/fe/webfastfood-sprint3/src/app/entity/Food.ts
import {ICategory} from './ICategory';
import {IOrderDetail} from './IOrderDetail';

export interface Food {
  fadId: number;
  fadName: string;
  fadImage: string;
  fadCode: string;
  fadPrice: number;
  // fadWaitTime: number;
  deleteFlag: boolean;
  category: ICategory;
  quantityPeopleOrder:number;
  orderDetails: IOrderDetail[];
  description:string;


}

File: clone-sprint2-3/fastfood/sprint3_backend_DiepLV/webfastfood/src/main/java/com/example/webfastfood/entity/food/Food.java
package com.example.webfastfood.entity.food;




import com.example.webfastfood.entity.oders.OrderDetail;
import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Food {
    @JsonBackReference(value = "items_orders")
    @OneToMany(mappedBy = "food")
    Set<OrderDetail> orderDetails;
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer fadId;
    private String fadName;
    private String fadImage;
    private boolean deleteFlag;
    private String fadCode;
    private double fadPrice;
    //    private Integer fadWaitTime;
    private Integer quantityPeopleOrder;

    @ManyToOne(targetEntity = Category.class)
    @JoinColumn(name = "category_id", referencedColumnName = "categoryId")
    private Category category;
}